composer/utils/inference.py

```diff
@@ -0,0 +1,175 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Inference-related utility functions for model export and optimizations.
+
+Used for exporting models into various formats such ONNX, torchscript etc. and apply optimizations such as fusion.
+"""
+
+import contextlib
+import copy
+import logging
+import os
+import tempfile
+from typing import Any, Callable, Optional, Sequence, Union
+
+import torch
+import torch.nn as nn
+
+from composer.utils import dist
+from composer.utils.checkpoint import download_checkpoint
+from composer.utils.iter_helpers import ensure_tuple
+from composer.utils.object_store import ObjectStore
+from composer.utils.string_enum import StringEnum
+
+log = logging.getLogger(__name__)
+
+__all__ = ['export_for_inference', 'ExportFormat']
+
+
+class ExportFormat(StringEnum):
+    """Enum class for the supported export formats.
+
+    Attributes:
+        torchscript: Export in "torchscript" format.
+        onnx:  Export in "onnx" format.
+    """
+    TORCHSCRIPT = 'torchscript'
+    ONNX = 'onnx'
+
+
+def is_model_deepspeed(model) -> bool:
```

Comment:
Could we move this into a shared helper?

Reply:
Added a standalone functions and replaced all uses.